### ISSUE TITLE
New Provider import Types

### ISSUE THREAD

**Issue by ndom91** [nextauthjs/next-auth#2515]
### Question 💬

Importing providers via the new method, i.e.

```js
import GoogleProvider from 'next-auth/providers/google'
```

Requires some updating of the TS types.

### How to reproduce ☕️

```js
import GoogleProvider from 'next-auth/providers/google'
```

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by balazsorban44** [comment]
We will need to generate some module declarations, similarly to how we generated the `src/providers/index.js` file.

**Comment by balazsorban44** [comment]
@ndom91 here is a version where I attempted to fix all these, have a look: `next-auth@0.0.0-pr.2518.7bdfc7da`

**Comment by balazsorban44** [comment]
Reviewed with @lluia
